internal/pkg/cli/svc_logs.go

```diff
@@ -65,26 +64,30 @@ func newSvcLogOpts(vars svcLogsVars) (*svcLogsOpts, error) {
 	if err != nil {
 		return nil, fmt.Errorf("connect to deploy store: %w", err)
 	}
-	return &svcLogsOpts{
+	opts := &svcLogsOpts{
 		svcLogsVars: vars,
 		w:           log.OutputWriter,
 		configStore: configStore,
 		deployStore: deployStore,
 		sel:         selector.NewDeploySelect(vars.prompt, configStore, deployStore),
-		initCwLogsSvc: func(o *svcLogsOpts, envName string) error {
-			env, err := o.configStore.GetEnvironment(o.AppName(), envName)
-			if err != nil {
-				return fmt.Errorf("get environment: %w", err)
-			}
-			sess, err := sessions.NewProvider().FromRole(env.ManagerRoleARN, env.Region)
-			if err != nil {
-				return err
-			}
-			o.cwlogsSvc[env.Name] = cloudwatchlogs.New(sess)
-			return nil
-		},
-		cwlogsSvc: make(map[string]cwlogService),
-	}, nil
+	}
+	opts.initLogsSvc = func() error {
```

Comment:
good question! i think this is just because in this way we can have a better method signature as `func() error` instead of `func(o *svcLogsOpts) error`. Other than that there's no benefit for doing this.